### PR TITLE
[Backend] Pickup Request Step 1

### PR DESCRIPTION
**Pull request by arielin3**
Trello task: https://trello.com/c/Lbc7PzNr/24-delivery-backend-1-recurring-pickup-request

**Step 1 in pickup request backend: write request to database.**

**Changes:**

- `PageContainer`: query for the `DonatingAgency` object from DB if the user logged in is `DonatingAgencyMember`
- `RecurringPickupRequest`:
  - fill in code to query all DA members, RAs, and DGs from DB to populate the dropdown lists
  - create `DeliveryRequest` object and write to DB
- Other minor changes:
  - added `moment` module
  - move getting weekday from date to a utils file
  - make repeats & primary contact dropdown lists required in the form
  - minor fixes and style changes

**Test:** tested by creating requests through UI and examine the database

**TODOs:**
- Populate `PickupSummary` with data from db
- Add some sort of "Request placed"/"Notifications sent" message after hitting `Confirm`

### CODE REVIEW

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 1]
```diff
@@ -130,29 +171,34 @@ class RecurringPickupRequest extends Component {
                 endTime: event.target.endTime.value,
                 primaryContact: event.target.primaryContact.value,
                 notes: event.target.notes.value,
+                umbrella: this.props.donatingAgency.umbrella,
+                donatingAgency: this.props.account.agency,
+                requester: this.props.account.name,
                 receiveingAgency: {
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: typo

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 2]
```diff
@@ -1,39 +1,71 @@
 import React, {Component} from 'react';
+
+import firebase, { accountsRef } from '../../FirebaseConfig.js';
 import { RequestRepeatType, RequestDurationType } from '../../Enums.js';
+import { getWeekdayFromDateString } from '../../Utils.js';
+
```
Comment (huangjoyce3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: nit-pick: delete space in lines 2 and 6

**Comment by jkbach** [comment]
So I just repro'd this one, everything is populated in the db ok :)   I think it's a little odd that I can keep clicking `Confirm` and it will keep writing to the db. Ideally, we'd disable the button after the first request is sent, and have a handler for when the response comes back so we can close the pop-up submit dialog and give a helpful message like "your request has been placed"

**Comment by arielin3** [comment]
@jkbach that was one of the TODOs "What happens after clicking on `Confirm`?" lol but I actually just made some changes that'd close the popup and clear the form after hitting `Confirm`. I agree we should have something that says "request has been placed" or "notifications sent to receiving agencies" or something like that, but i think that can be a low priority for now.

File: src/Enums.js [hunk 3]
```diff
@@ -2,6 +2,11 @@
 /*global exports:true*/
 /*eslint no-undef: "error"*/
 
+exports.DateTimeFormat = {
+    DATE: 'YYYY-MM-DD',  // '2018-03-24'
+    TIME: 'HH:mm',  // '15:30'
```
[the remark this answers is not in the document]
Comment (arielin3) [reply src/Enums.js]: nope! the `HH` format is 24hr format

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 4]
```diff
@@ -87,39 +120,49 @@ class RecurringPickupRequest extends Component {
...
-    }
-
+    // When "Done" is clicked on the request form
     createRequest(event) {
-        
-        if(!this.handleValidation()){ 
-            event.preventDefault();
+        event.preventDefault();
+        if (!this.handleValidation()) {
             alert('Form has errors');
```
[the remark this answers is not in the document]
Comment (arielin3) [reply src/PageContent/RequestPickup/RecurringPickupRequest.js]: the dialog is handled by the browser i believe, but yes i agree that alerts are bad lol i'll add a TODO for joyce to change this

**Pull request merged**